File: api/src/main/java/org/pmiops/workbench/utils/CommonMappers.java
package org.pmiops.workbench.utils;

import java.sql.Timestamp;
import org.mapstruct.Mapper;

@Mapper(componentModel = "spring")
public class CommonMappers {

  public static Long timestamp(Timestamp timestamp) {
    if (timestamp != null) {
      return timestamp.getTime();
    }

    return null;
  }

  public static Timestamp timestamp(Long timestamp) {
    if (timestamp != null) {
      return new Timestamp(timestamp);
    }

    return null;
  }

}

File: api/src/main/java/org/pmiops/workbench/utils/CommonMappers.kt
package org.pmiops.workbench.utils

import java.sql.Timestamp
import org.mapstruct.Mapper

@Mapper(componentModel = "spring")
object CommonMappers {

    fun timestamp(timestamp: Timestamp?): Long? {
        return timestamp?.time

    }

    fun timestamp(timestamp: Long?): Timestamp? {
        return if (timestamp != null) {
            Timestamp(timestamp)
        } else null

    }
}
